### ISSUE TITLE
Firebase database and swr

### ISSUE THREAD

**Issue by DoT214** [vercel/swr#249]
Question : Is it a good strategy to use swr with firebase ?

**Comment by sergiodxa** [comment]
You could use the Firebase JS client as fetcher, probably with a wrapper, but I’m not sure how useful it would be 🤔

**Comment by illuminist** [comment]
Realtime database and Firestore has their own optimized caching strategy so it's better to let them handle their own data.

**Comment by danoc** [comment]
Highly recommend checking out [swr-firebase](https://github.com/nandorojo/swr-firestore). I'm playing around with it now and it seems very well done.

**Comment by shuding** [comment]
@danoc wow I didn't know about that lib, it looks truly amazing. Thank you!

**Comment by optimista** [comment]
I did some research whether to implement [swr-firestore](https://github.com/nandorojo/swr-firestore) or not for an own personal project. This is what I concluded and some of the distinctive features which I found. Happy to discuss! I'm far from understanding everything.

[swr-firestore](https://github.com/nandorojo/swr-firestore)

- can listen to e.g. document's get()s across screens (getting same document on other screen without swr won't update first one - see: https://www.reddit.com/r/Firebase/comments/he3qtg/firestore_hooks_library_for_react_useswr/)
- supports multiple order-bys, limits (easier writing of queries)
- has a pagination model
- supports React Suspense

As far as I did my research, the other realtime features can be resolved with e.g. `onSnapshot` listeners of firestore realtime database which provides its own caching system, even ordinary `get()`s do cache, maybe without that screen update though. This made me conclude that even though it can be nice to have, the costs of implementation and upkeeping might be just a little higher than dealing just with the raw firestore client. Still - something to keep an eye on!
